File: Backend/Felino.Api/Domain/Entities/Campaign.cs
using System.ComponentModel.DataAnnotations;

namespace Felino.Api.Domain.Entities
{
    public class Campaign
    {
        public int Id { get; set; }

        [Required]
        [MaxLength(100)]
        public string Title { get; set; } = "";

        [Required]
        [MaxLength(200)]
        public string Body { get; set; } = "";

        public string? ImageUrl { get; set; }
        
        public string? AltText { get; set; }

        public DateTime StartDate { get; set; }

        public DateTime EndDate { get; set; }
    }
}

File: Frontend/client/src/types/campaign.ts
export type CampaignDto = {
  id: number;
  title: string;
  body: string;
  imageUrl?: string | null;
  altText?: string | null;
  startDate: string;
  endDate: string;
};

export type Campaign = {
  id: number;
  title: string;
  body: string;
  image?: string;
  altText?: string;
  startDate: string;
  endDate: string;
  status: "active" | "upcoming" | "expired";
};

export type CreateCampaignRequest = {
  title: string;
  body: string;
  imageUrl?: string | null;
  altText?: string | null;
  startDate: string;
  endDate: string;
};

export type UpdateCampaignRequest = CreateCampaignRequest;